.github/workflows/test.yml

```diff
@@ -0,0 +1,55 @@
+name: 'Test'
+on:
+  pull_request:
+  workflow_dispatch:
+# Stop in progress workflows on the same branch and same workflow to use latest committed code
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test-pr:
+    name: 'Test PR'
+    runs-on: [self-hosted, linux, normal]
+    steps:
+      - name: 'Check out code'
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: 'Set up Docker'
+        run: |
+          set -euxo pipefail
+
+          cd deps/k
+          git fetch --no-recurse-submodules origin 'refs/tags/*:refs/tags/*'
+          K_VERSION="$(git tag --points-at HEAD | cut --characters=2- | head -n 1)"
+          cd ../..
+
+          KAVM_COMMIT="$(git rev-parse --short=7 HEAD)"
+
+          docker build . --file Dockerfile --build-arg K_VERSION=${K_VERSION} --tag runtimeverification/avm-semantics-ci:${KAVM_COMMIT} --build-arg USER_ID=$(id -u) --build-arg GROUP_ID=$(id -g)
+
+          docker run --name avm-semantics-ci --rm -it --detach --workdir /opt/workspace --user $(id -u):$(id -g) -v ${HOME}:${HOME} -v "$(pwd):/opt/workspace" -v "/etc/passwd:/etc/passwd:ro" -v "/etc/group:/etc/group:ro" runtimeverification/avm-semantics-ci:${KAVM_COMMIT}
```

Comment:
@F-WRunTime are we not supposed to do `--rm` now? In favor of the explicit `docker stop ...` at the end?

Reply:
We should use the --rm in favor of calling `docker rm` 
The --rm flag will cause clean up automatically when `docker stop` is called.